### PR TITLE
docs: refactor README content

### PR DESCRIPTION
**Pull request by deadprogram**
This PR refactors README.md to avoid duplication with information already on the web site, and to reorder to make it easier for new users. It also moves a few details over to the CONTRIBUTING.md document.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,179 +1,109 @@
...
 
--- Rob Pike, [GopherCon 2014 Opening Keynote](https://www.youtube.com/watch?v=VoS7DsT1rdM&feature=youtu.be&t=2799)
-
-TinyGo is a project to bring Go to microcontrollers and small systems with a
-single processor core. It is similar to [emgo](https://github.com/ziutek/emgo)
-but a major difference is that I want to keep the Go memory model (which implies
-garbage collection of some sort). Another difference is that TinyGo uses LLVM
-internally instead of emitting C, which hopefully leads to smaller and more
-efficient code and certainly leads to more flexibility.
+It uses the [Go compiler toolchain](https://golang.org/pkg/go/) alongside [LLVM](http://llvm.org) to provide an alternative way to compile programs written in the Go programming language.
```
Comment (aykevl) [inline README.md]: That's only partially true. It does not really reuse anything from the Go compiler toolchain, but reuses the set of libraries used by all the Go tools (in and out of the `go` command).

File: README.md [hunk 2]
```diff
@@ -1,179 +1,109 @@
...
 
-My original reasoning was: if [Python](https://micropython.org/) can run on
-microcontrollers, then certainly [Go](https://golang.org/) should be able to and
-run on even lower level micros.
-
-Example program (blinky):
+Here is an example program that blinks the built-in LED when run directly on an Arduino Uno microcontroller:
```
Comment (aykevl) [inline README.md]: This code is actually supported on many different boards, basically all boards that have a built-in LED. What about this?

> Here is an example program that blinks the built-in LED when run directly on any supported board with onboard LED:

**Comment by deadprogram** [comment]
OK updates pushed.

**Comment by deadprogram** [comment]
Now merging, thanks for the feedback.

**Pull request merged**